ibis/expr/operations.py

```diff
@@ -2692,3 +2800,17 @@ def root_tables(self):
     def _make_expr(self):
         dtype = rlz.highest_precedence_dtype(self.values)
         return ir.ListExpr(self, dtype=dtype)
+
+
+# GEOMETRIC OPERATIONS
+
+class Distance(ValueOp):
+    """
+    Calculates distance in meters between two WGS-84 positions.
+
+    """
+    from_lon = Arg(rlz.column(rlz.numeric))
+    from_lat = Arg(rlz.column(rlz.numeric))
+    to_lon = Arg(rlz.column(rlz.numeric))
+    to_lat = Arg(rlz.column(rlz.numeric))
+    output_type = rlz.shape_like('from_lon', dt.float)
```

Comment:
@xmnlab Can't be any of the arguments scalar like?

Reply:
@kszucs yes any of the args could be a scalar or a column, should I changed to just `rlz.numeric` ?

Reply:
@xmnlab Then use the following definition:

```python
 from_lat = Arg(rlz.numeric)
 to_lon = Arg(rlz.numeric)
 to_lat = Arg(rlz.numeric)
 output_type = rlz.shape_like('args', dt.float64)
```

This way, if any of the arguments is column like, then the return type will be columnar shaped too.